File: BallistiCore_app/frontend/src/api/exports.js
import api from './client'

const ZIP = 'application/zip'

// Download the full data export (Excel + CSV bundle + PDF summary) as one ZIP.
export const downloadFullExport = async () => {
  const res = await api.get('/api/export/all', { responseType: 'blob' })
  const disposition = res.headers['content-disposition'] || ''
  const match = disposition.match(/filename="?([^"]+)"?/)
  const filename = match ? match[1] : 'BallistiCore_Export.zip'
  const url = URL.createObjectURL(new Blob([res.data], { type: ZIP }))
  const a = document.createElement('a')
  a.href = url
  a.download = filename
  document.body.appendChild(a)
  a.click()
  a.remove()
  URL.revokeObjectURL(url)
}

File: BallistiCore_app/backend/app/routers/exports.py
from fastapi import APIRouter, Depends
from fastapi.responses import Response
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.core.auth import require_admin
from app.services import exports as svc

router = APIRouter(prefix="/api/export", tags=["Export"])


@router.get("/all")
def export_all_data(db: Session = Depends(get_db), current_user=Depends(require_admin)):
    """Export the complete system dataset (Excel + CSV bundle + PDF summary) as a
    single downloadable ZIP.

    Admin-only (require_admin == super admin) because the data includes ID
    numbers, SAPS competency numbers and PSIRA numbers. The service logs each
    export with the requesting user for audit purposes.
    """
    data, filename = svc.build_full_export(db, generated_by=current_user.username)
    return Response(
        content=data,
        media_type="application/zip",
        headers={"Content-Disposition": f'attachment; filename="{filename}"'},
    )
